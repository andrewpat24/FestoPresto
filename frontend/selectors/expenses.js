const getVisibleExpenses = (
  expenses,
  /* filters:*/ { text, sortBy, startDate, endDate }
) => {
  return expenses
    .filter(expense => {
      const startDateMatch = !startDate || expense.createdAt >= startDate;
      const endDateMatch = !endDate || expense.createdAt <= endDate;
      const textMatch = expense.description.toLowerCase().includes(text);
      return startDateMatch && endDateMatch && textMatch;
    })
    .sort((a, b) => {
      if (sortBy === "date") {
        return a.createdAt < b.createdAt ? 1 : -1;
      } else if (sortBy === "amount") {
        return a.amount < b.amount ? 1 : -1;
      }
    });
};

export default getVisibleExpenses;
