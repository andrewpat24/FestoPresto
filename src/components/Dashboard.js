import React from "react";
// Components
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";
import ExpensesSummary from "./ExpensesSummary";

const Dashboard = () => (
  <section component="Dashboard">
    <ExpensesSummary />
    <ExpenseListFilters />
    <ExpenseList />
  </section>
);

export default Dashboard;
