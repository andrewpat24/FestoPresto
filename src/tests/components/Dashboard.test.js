import React from "react";
import { shallow } from "enzyme";
import Dashboard from "../../components/Dashboard";
import expenses from "../fixtures/expenses";

test("Should render dashboard", () => {
  const wrapper = shallow(<Dashboard />);
  expect(wrapper).toMatchSnapshot();
});
