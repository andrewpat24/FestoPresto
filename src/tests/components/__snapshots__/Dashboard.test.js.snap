// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render dashboard 1`] = `
<section
  component="Dashboard"
>
  <Connect(ExpensesSummary) />
  <Connect(ExpenseListFilters) />
  <Connect(ExpenseList) />
</section>
`;
